### DISCUSSION TITLE
OpenPdf 2.0.2 and 2.0.3 fail in Document.newPage()

### DISCUSSION THREAD

**Discussion by arnthom** [LibrePDF/OpenPDF#1248] [Q&A]
Hi everybody!
I use OpenPDF as a tool to create PDF documentation of my software project.
After finishing a page, an event is used to create header and footer.
Recently I tried a new version of OpenPdf and Java 17.

With OpenPdf after 1.3.43 an error occurs:
`Caused by: java.lang.NoSuchMethodError: void com.lowagie.text.pdf.PdfPTable.addCell(com.lowagie.text.pdf.PdfPCell)`

This is rather strange, because this method exists in older and newer versions of OpenPdf.

When I try to check this in the Eclipse project, I try to open the implemantation of Document.newPage() and get
`Problems finding implemantations.
Reason:
An error occurred while searching for implemantations of "new Page". See error log for details.
META-INF.versions.17.com.lowagie.text [in...] does not exist`

Can you reproduce this problem?
Do you need more informations?
Is the a workaround?

Thanks!

btw: I used the title "OpenPdf 2.0.2 and ..." before finding, that 1.4.2 as well has this problem.
Maybe there is abug in all versions after 1.3.43 or some incompatibility exists in the configuration I use.

**Comment by arnthom** [comment]
Any ideas or tipps, where I could search for a workaround?

**Reply by alesky78** [reply]
i use exactly the version <com.github.librepdf>1.3.43</com.github.librepdf>, 

but I trust that the  Document.newPage() works correctly as I imagine it is an intensively used method

I suggest you to create a project wiht a very small unit test to create a proper pdf with just one paragraph and then call Document.newPage() and test wiht differente versione to prove that what you think is correct

moreover i use (again wiht version 1.3.43) PdfPTable.addCell wherever wihtout any problem

looking at this error,
Caused by: java.lang.NoSuchMethodError: void com.lowagie.text.pdf.PdfPTable.addCell(com.lowagie.text.pdf.PdfPCell)

may be you use the library at compile time but do not export it at run time and therefore it is not part of the classpath

**Reply by arnthom** [reply]
Hello alesky78!

With 1.3.43 it works as expected.
Version 1.4.2 and 2.0.x lead to this error.

Maybe it is just some problem with the library and not with the method itself.
The manifest file has changed a little bit, but I cannot see, if that has something to do with the error I get.


@alesky78 : can you change to 1.4.2 or 2.x and everything works fine?

**Comment by arnthom** [comment]
Still a very strange thing ...
Now I can test within my development setting (Win7) and OpenPdf 2.0.3 works smoothly, but when deploying the software to a Win11 test machine, only OpenPdf 1.3.43 produces the resulting pdf document. OpenPdf 1.4.2 and later have this error ...

Why does this error occur?
`Caused by: java.lang.NoSuchMethodError: 'void com.lowagie.text.pdf.PdfPTable.addCell(com.lowagie.text.pdf.PdfPCell)'`
The method still exists in OpenPdf 2.0.3
Maybe the OS (Windows 11) is relevant?

**Comment by andreasrosdalw** [comment]
Here's links to the source code of the mentioned methods, which can be used to understand this issue better:


PdfTable.addCell():
https://github.com/LibrePDF/OpenPDF/blob/master/openpdf/src/main/java/com/lowagie/text/pdf/PdfPTable.java#L500

Document.newPage():
https://github.com/LibrePDF/OpenPDF/blob/master/openpdf/src/main/java/com/lowagie/text/Document.java#L419

**Comment by andreasrosdalw** [comment]
Note that PdfTable.addCell changed method signature.

https://github.com/LibrePDF/OpenPDF/issues/837

https://github.com/LibrePDF/OpenPDF/pull/844
